components/server/src/user/user-service.ts

```diff
@@ -425,41 +384,6 @@ export class UserService {
         await this.userDb.storeSingleToken(candidate, token);
     }
 
-    async updateUserEnvVarsOnLogin(user: User, envVars?: UserEnvVarValue[]) {
```

Comment:
Is this no longer used? It doesn't seem to be related to the TOS flow

Reply:
ToS flow payload used to capture these. It didn't make sense to move the call site of this when moving the remainings of the `TosFlow` shape, which is now a plain result type of a function and doesn't represent a state of the session. Short: it's dead code. There is no use case for deriving/setting env vars on logins, which was a SH special anyways.

Reply:
Hey @AlexTugarev! While that may as well be the case for gitpod.io, we actually do need to set an environment variable or two for some users, in some environments (installs), e.g. in a test env.

With this code removed, could you suggest/recommend a flow or a place in code where would be the best to set user env vars? Ideally a place that's relatively safe/robust going forward (won't be yanked anytime soon). Thanks!

/cc @charleswhchan